Cubical/Foundations/SIP.agda

```diff
@@ -92,122 +76,23 @@ SNS-≡→SNS-PathP {S = S} ι θ {A = A} {B = B} e = EquivJ P C e (str A) (str
-   ≡⟨ (λ i j → ( p j
-               , transportTransport⁻ (lem S A B p) q i j)) ⟩
-     r ∎
-
-
--- The trickier direction:
-sip⁻-sip : {S : Type ℓ₁ → Type ℓ₂} {ι : StrIso S ℓ₃} (θ : SNS-PathP S ι)
-           (A B : TypeWithStr ℓ₁ S) (r : A ≃[ ι ] B)
-         → sip⁻ θ A B (sip θ A B r) ≡ r
-sip⁻-sip {S = S} θ A B (e , p) =
-    sip⁻ θ A B (sip θ A B (e , p))
-  ≡⟨ refl ⟩
-    pathToEquiv (ua e) , invEq (θ (pathToEquiv (ua e))) (f⁺ p')
-  ≡⟨ (λ i → pathToEquiv-ua e i , invEq (θ (pathToEquiv-ua e i)) (pth' i)) ⟩
-    e , invEq (θ e) (f⁻ (f⁺ p'))
-  ≡⟨ (λ i → e , invEq (θ e) (transportTransport⁻ (lem S A B (ua e)) p' i)) ⟩
-    e , invEq (θ e) (θ e .fst p)
-  ≡⟨ (λ i → e , (secEq (θ e) p i)) ⟩
-    e , p ∎
-  where
-  p' : PathP (λ i → S (ua e i)) (str A) (str B)
-  p' = θ e .fst p
-
-  f⁺ : PathP (λ i → S (ua e i)) (str A) (str B)
-     → PathP (λ i → S (ua (pathToEquiv (ua e)) i)) (str A) (str B)
-  f⁺ = transport (λ i → PathP (λ j → S (ua-pathToEquiv (ua e) (~ i) j)) (str A) (str B))
-
-  f⁻ : PathP (λ i → S (ua (pathToEquiv (ua e)) i)) (str A) (str B)
-     → PathP (λ i → S (ua e i)) (str A) (str B)
-  f⁻ = transport (λ i → PathP (λ j → S (ua-pathToEquiv (ua e) i j)) (str A) (str B))
-
-  -- We can prove the following as in sip∘pis-id, but the type is not
-  -- what we want as it should be "cong ua (pathToEquiv-ua e)"
-  pth : PathP (λ j → PathP (λ k → S (ua-pathToEquiv (ua e) j k)) (str A) (str B))
-              (f⁺ p') (f⁻ (f⁺ p'))
-  pth i = transp (λ k → PathP (λ j → S (ua-pathToEquiv (ua e) (i ∧ k) j)) (str A) (str B))
-                 (~ i) (f⁺ p')
-
-  -- So we build an equality that we want to cast the types with
-  casteq : PathP (λ j → PathP (λ k → S (ua-pathToEquiv (ua e) j k)) (str A) (str B))
-                 (f⁺ p') (f⁻ (f⁺ p'))
-         ≡ PathP (λ j → PathP (λ k → S (cong ua (pathToEquiv-ua e) j k)) (str A) (str B))
-                 (f⁺ p') (f⁻ (f⁺ p'))
-  casteq i = PathP (λ j → PathP (λ k → S (eq i j k)) (str A) (str B)) (f⁺ p') (f⁻ (f⁺ p'))
-    where
-    -- This is where we need the half-adjoint equivalence property
-    eq : ua-pathToEquiv (ua e) ≡ cong ua (pathToEquiv-ua e)
-    eq = sym (uaHAEquiv (typ A) (typ B) .snd .com e)
-
-  -- We then get a term of the type we need
-  pth' : PathP (λ j → PathP (λ k → S (cong ua (pathToEquiv-ua e) j k)) (str A) (str B))
-               (f⁺ p') (f⁻ (f⁺ p'))
-  pth' = transport (λ i → casteq i) pth
-
--- Finally package everything up to get the cubical SIP
-SIP : {S : Type ℓ₁ → Type ℓ₂} {ι : StrIso S ℓ₃}
-      (θ : SNS-PathP S ι) (A B : TypeWithStr ℓ₁ S)
-    → A ≃[ ι ] B ≃ (A ≡ B)
-SIP {S = S} {ι} θ A B = isoToEquiv f
-  where
-  f : Iso (A ≃[ ι ] B) (A ≡ B)
-  Iso.fun f      = sip θ A B
-  Iso.inv f      = sip⁻ θ A B
-  Iso.rightInv f = sip-sip⁻ θ A B
-  Iso.leftInv f  = sip⁻-sip θ A B
 
+  SIP : A ≃[ ι ] B ≃ (A ≡ B)
+  SIP =
+    isoToEquiv (compIso (Σ-cong-iso (equivToIso (invEquiv univalence)) (equivToIso ∘ θ)) ΣPathIsoPathΣ)
+
+  sip : A ≃[ ι ] B → A ≡ B
```

Comment:
It this definitionally the same as the old map? Maybe we could keep the old map and just construct the isEquiv part of the above proof so that it is clear what the computational behavior of `sip` is?

Reply:
What do you think?

Reply:
I think it's definitionally the same map, I can expand it out.